packages/perseus/src/widgets/interactive-graphs/graph-locked-layer.tsx

```diff
@@ -22,6 +24,7 @@ const GraphLockedLayer = (props: Props) => {
                         break;
                     case "line":
                         Figure = LockedLine;
+                        otherProps = {range: props.range};
```

Comment:
Could we move the declaration of `otherProps` inside the `map` callback, next to Figure?

```ts
{lockedFigures.map((figure, index) => {
    let Figure;
    let otherProps;
```

The way it is now, the `otherProps` will persist between calls to the `map` callback. This means that if the first element in `lockedFigures` is a line, and the second is a point, the `range` prop will get mistakenly added to the point. Probably not a huge deal, but could create headaches down the line.

I'm also kind of surprised TypeScript accepts this, and I wonder if it's just throwing up its hands and implicitly typing `otherProps` as `any` or something.

An alternative would be to render the figure inside the `switch`:

```tsx
{lockedFigures.map((figure, index) => {
    switch (figure.type) {
        case "point":
            return <LockedPoint key={`point-${index}`} {...figure} />;
            break;
        case "line":
            return <LockedLine key={`line-${index}`} range={props.range} {...figure} />;
            break;
        default:
            /**
             * Devlopment-time future-proofing: This should
             * fail during type-checking if we add a new locked
             * shape type and forget to handle it in any other
             * switch case here.
             */
            throw new UnreachableCaseError(figure);
    }
})}
```

Reply:
oh yeah rendering directly inside the `swtich` makes way more sense. The current implementation is left over from how it was done pre-refactor where it was super long awkward to put in a swtich. Thanks!